### PR TITLE
(feat): Enable flexible scaling in storage cluster deployment

### PR DESCRIPTION
**Pull request by afreen23**
- implements https://issues.redhat.com/browse/RHSTOR-1399
 - moves arbiter form group to generic file
**Internal mode**
Screenshot from 2020-12-04 14-17-13
Screenshot from 2020-12-04 14-17-22

### CODE REVIEW

**Comment by afreen23** [comment]
Merge after https://github.com/openshift/console/pull/7376

File: frontend/packages/ceph-storage-plugin/src/utils/install.ts [hunk 1]
```diff
@@ -80,3 +85,36 @@ export const shouldDeployAsMinimal = (cpu: number, memory: number, nodesCount: n
...
+  const scNodeNames = getAssociatedNodes(pvs);
+  const filteredNodes: NodeKind[] = nodesWithoutTaints(nodesData);
+  const tableData: NodeKind[] = filteredNodes.filter(
+    (node: NodeKind) =>
+      scNodeNames.includes(getName(node)) ||
+      scNodeNames.includes(node.metadata.labels?.['kubernetes.io/hostname']),
+  );
+  const uniqZones: Set<string> = new Set(filteredNodes.map((node) => getZone(node)));
+  const uniqSelectedNodesZones: Set<string> = new Set(tableData.map((node) => getZone(node)));
+  if ([...uniqZones].length < 3) return false;
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/utils/install.ts:116]:
```suggestion
  if (uniqZones.size < 3) return false;
```

File: frontend/packages/ceph-storage-plugin/src/utils/install.ts [hunk 2]
```diff
@@ -80,3 +85,36 @@ export const shouldDeployAsMinimal = (cpu: number, memory: number, nodesCount: n
...
+  if ([...uniqSelectedNodesZones].length !== 2) return false;
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/utils/install.ts:117]:
```suggestion
  if (uniqSelectedNodesZones.size !== 2) return false;
```

File: frontend/packages/ceph-storage-plugin/src/utils/common-ocs-install-el.tsx [hunk 3]
```diff
@@ -38,6 +38,8 @@ enum ValidationType {
   'ENCRYPTION' = 'ENCRYPTION',
   'REQUIRED_FIELD_KMS' = 'REQUIRED_FIELD_KMS',
   'NETWORK' = 'NETWORK',
+  'INTERNALFLEXIBLESCALING' = 'INTERNALFLEXIBLESCALING',
+  'BAREMETALFLEXIBLESCALING' = 'BAREMETALFLEXIBLESCALING',
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/utils/common-ocs-install-el.tsx:42]: `INTERNAL_FLEXIBLE_SCALING & BAREMETAL_FLEXIBLE_SCALING`
More readable.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-request-data.ts [hunk 4]
```diff
@@ -71,17 +71,23 @@ export const labelNodes = (selectedNodes: NodeKind[]): Promise<NodeKind>[] => {
   );
 };
 
+// export enum Configuration {
+//   'Arbiter' = 'Arbiter',
+//   'FlexibleScaling' = 'FlexibleScaling',
+// }
+
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-request-data.ts:78]: This is for??

**Review by bipuladh — CHANGES_REQUESTED** [review]

**Review by bipuladh — APPROVED** [review]
/lgtm

**Comment by cloudbehl** [comment]
/hold

 for https://github.com/openshift/console/pull/7376

**Comment by cloudbehl** [comment]
/hold cancel

@afreen23 can you rebase!

**Pull request merged**